src/overlays/actors/ovl_En_Dinofos/z_en_dinofos.c

```diff
@@ -114,144 +257,1279 @@ static DamageTable D_8089E314 = {
+                }
+            }
+
+            if (this->actor.cutscene != -1) {
+                func_801A2ED8();
+            }
+        }
+
+        func_8089ACEC(this, globalCtx);
+        func_8089AD70(this);
+        if (globalCtx->sceneNum == SCENE_MITURIN) {
+            globalCtx->envCtx.unk_C3 = 255;
+        }
+
+        this->colliderQuad.base.atFlags &= ~5;
+        this->colliderJntSph.base.atFlags &= ~1;
+        if (this->actor.colChkInfo.damageEffect == 3) {
+            func_8089AC70(this);
+            if (this->actor.colChkInfo.health == 0) {
+                this->unk_290 = 3;
+                this->colliderJntSph.base.acFlags &= ~1;
+            }
+            func_8089C784(this);
+            return 1;
+        }
+
+        if (this->actor.colChkInfo.damageEffect == 1) {
+            this->unk_290 = 40;
+            func_800BCB70(&this->actor, 0, 255, 0, 40);
+            Audio_PlayActorSound2(&this->actor, NA_SE_EN_COMMON_FREEZE);
+            func_8089C784(this);
+            return 1;
+        }
+
+        if (this->actor.colChkInfo.damageEffect == 5) {
+            this->unk_290 = 40;
+            func_800BCB70(&this->actor, 0, 255, 0, 40);
+            this->unk_2B4 = 0.55f;
+            this->unk_2B0 = 2.0f;
+            this->unk_28A = 31;
+            Audio_PlayActorSound2(&this->actor, NA_SE_EN_COMMON_FREEZE);
+            func_8089C784(this);
+            return 1;
+        }
+
+        if (this->actor.colChkInfo.damageEffect == 2) {
+            this->unk_2B0 = 4.0f;
+            this->unk_2B4 = 0.55f;
+            this->unk_28A = 0;
+        } else if (this->actor.colChkInfo.damageEffect == 4) {
+            this->unk_2B0 = 4.0f;
+            this->unk_2B4 = 0.55f;
+            this->unk_28A = 20;
+            Actor_Spawn(&globalCtx->actorCtx, globalCtx, ACTOR_EN_CLEAR_TAG,
+                        this->colliderJntSph.elements[i].info.bumper.hitPos.x,
+                        this->colliderJntSph.elements[i].info.bumper.hitPos.y,
+                        this->colliderJntSph.elements[i].info.bumper.hitPos.z, 0, 0, 0, CLEAR_TAG_LARGE_LIGHT_RAYS);
+        }
+        func_8089C87C(this, i);
+        return 1;
+    }
+
+    return 0;
+}
+
+void EnDinofos_Update(Actor* thisx, GlobalContext* globalCtx2) {
+    EnDinofos* this = THIS;
+    GlobalContext* globalCtx = globalCtx2;
+    s32 pad;
+    Vec3f sp30;
+
+    if (this->actionFunc != func_8089C7B8) {
+        func_8089A900(this);
+    }
+
+    if (!func_8089D60C(this, globalCtx) && (this->colliderQuad.base.atFlags & 4)) {
+        func_8089C690(this);
+    }
+
+    this->actionFunc(this, globalCtx);
+    Actor_SetVelocityAndMoveYRotationAndGravity(&this->actor);
+    Actor_UpdateBgCheckInfo(globalCtx, &this->actor, 25.0f, 30.0f, 60.0f, 0x5D);
+    if (this->actionFunc != func_8089C7B8) {
+        if ((this->actor.yDistToWater > 0.0f) && (this->actor.yDistToWater < 10.0f)) {
+            if (!((globalCtx->gameplayFrames & 3) & 1)) {
```

Comment:
```suggestion
            if (!((globalCtx->gameplayFrames % 4) % 1)) {
```
If either of those match. & vs % can be finicky sometimes.